SMLHelper/Options/LabelAttribute.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+namespace SMLHelper.V2.Options
+{
+    /// <summary>
+    /// Attribute used to specify the label to display for the given field in the mod's options menu.
+    /// </summary>
+    /// <remarks>
+    /// Can also be used to manually specify the order in which to display options in the menu, or to specify the Id.
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// using SMLHelper.V2.Json;
+    /// using SMLHelper.V2.Options;
+    /// 
+    /// [Menu("My Options Menu")]
+    /// public class Config : ConfigFile
+    /// {
+    ///     [Label("My Cool Slider", Id = "myCoolSlider")]
+    ///     public float MyCoolSlider;
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="MenuAttribute"/>
+    /// <seealso cref="Json.ConfigFile"/>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Method, AllowMultiple = false)]
+    public sealed class LabelAttribute : Attribute
+    {
+        /// <summary>
+        /// The label to use when displaying the field in the mod's options menu.
+        /// </summary>
+        public string Label { get; internal set; }
+
+        /// <summary>
+        /// The Id to be used for the field in the mod's option menu. If none is specified, one will be automatically generated when
+        /// your <see cref="Json.ConfigFile"/> is registered to the <see cref="Handlers.OptionsPanelHandler"/>. This means it will
+        /// change every time the game is launched, but is guaranteed to be unique. If you would like to specify an Id to use for
+        /// internal comparisons, you can do so here.
+        /// </summary>
+        public string Id { get; set; } = Guid.NewGuid().ToString().Replace('-', '_');
+
+        /// <summary>
+        /// The order in which to display fields in the mod's option menu, in ascending order. If none is specified, the order will be
+        /// automatically set.
+        /// </summary>
+        public int Order { get; set; } = i++;
+        private static int i = 0;
```

Comment:
This static counter here has me nervous.  
And given that you are still allowing `Order` to be changed via the `set`, I can only imagine that this will cause trouble down the line.

Reply:
So far it does the job nicely. If you want to set the order manually, it's easy enough to document that you need to set the order for all of the members.

As for whether it has knock-on effects in other mods, this has yet to really be tested extensively. Ideally, I would be able to use the `[CallerLineNumber]` as the order in the constructor and then it's a very easily implied order based on the order of declaration - however that won't work across assemblies and definitely not in this version of .NET.

My simple take is this:
If you're using auto-order, it's not guaranteed to be the same every time.
If you want guaranteed order, set it for every member. Easy.

Reply:
I'm more concerned about what happens when we get multiple mods using this method.  
I suppose that might not matter given that base on this code from `ConfigModOptions`, this is being used more as a relative ordering rather than an index.
```CS
public override void BuildModOptions()
{
    foreach (var entry in modOptionsMetadata
        .OrderBy(x => x.Value.ModOptionAttribute.Order)
        .ThenBy(x => x.Value.MemberInfo.Name))
```

Personally, I would have these ordered in the same way they appear in the class.  
`Type.GetProperties()` and `Type.GetFields()` will both return their respective members the the order they appear in the class.  
Don't know if that helps.

Reply:
> I'm more concerned about what happens when we get multiple mods using this method.
> I suppose that might not matter given that, based on this code from `ConfigModOptions`, this is being used more as a relative ordering rather than an index.
> 
> ```cs
> public override void BuildModOptions()
> {
>     foreach (var entry in modOptionsMetadata
>         .OrderBy(x => x.Value.ModOptionAttribute.Order)
>         .ThenBy(x => x.Value.MemberInfo.Name))
> ```
> 
> Personally, I would have these ordered in the same way they appear in the class.
> `Type.GetProperties()` and `Type.GetFields()` will both return their respective members the the order they appear in the class.
> Don't know if that helps.

Yeah the issue with that is that if you have both fields and properties (and currently the design is to allow for both), then we have to make an arbitrary decision on which get listed first... fields or properties. And I don't like that. I would prefer them both to be listed in original order.

And yes, I've considered the case of multiple mods using it. It doesn't matter that mod #2 and mod #5 both use it, bc the number is just a relative order - it is used for ordering only in the local scope of the menu, and is just ordered in terms of "the lowest number is listed first". Whether it starts at 0 or 1000000 in your menu, it doesn't matter.

For what it's worth, from my testing so far, this method *does* list them in the correct order - the order they appear in the class, as the JIT is going through the class, one by one, evaluating the `[Toggle]` and then three lines down, you have a `[Keybind]` and it's processing those. Where it becomes not guaranteed is when you don't specify those yourself - then it becomes, automatically, order of the ones you declared attributes, then order of properties, then order of fields. And I personally think that's good enough?